wandb/sdk/wandb_run.py

```diff
@@ -1456,39 +1457,78 @@ def _datatypes_callback(self, fname: str) -> None:
         files: FilesDict = dict(files=[(GlobStr(glob.escape(fname)), "now")])
         self._backend.interface.publish_files(files)
 
+    def _get_nested_value(self, d: Dict[str, Any], keys: List[str]) -> Any:
+        if len(keys) == 1:
+            return d[keys[0]]
+        return self._get_nested_value(d[keys[0]], keys[1:])
+
+    def _set_nested_value(
+        self, d: Dict[str, Any], keys: List[str], value: Any, split_table: bool
```

Comment:
this method could use a docstring, in particular to explain what `split_table` is

Reply:
Hey @timoffex! I have rewritten the tests and added docstrings. Let me know if you still want me to move the tests into unit tests for speed. Shouldn't be extremely complicated.